client/app/components/notification/notification.js

```diff
@@ -0,0 +1,36 @@
+import angular from "angular";
+import uiRouter from "angular-ui-router";
+import notificationComponent from "./notification.component";
+import Authentication from "../../services/authentication/authentication";
+import User from "../../services/user/user";
+
+let notificationModule = angular.module("notification", [
+  uiRouter,
+  Authentication,
+  User
+])
+
+.component("notification", notificationComponent)
+
+.config(($stateProvider, hookProvider) => {
+  "ngInject";
+  $stateProvider
+    .state("notification", {
+      parent: "main",
+      url: "/notification/user/{id:int}",
```

Comment:
Could you nest the notifications page in the user state?
So that the URL looks like `/user/{id:int}/notifications`? Then you also don't need to resolve the user data again.

You can do this by setting `parent` appropriately.